src/server/protocol.ts

```diff
@@ -2121,7 +2135,7 @@ namespace ts.server.protocol {
         text: string;
 
         /**
-         * The category of the diagnostic message, e.g. "error" vs. "warning"
+         * The category of the diagnostic message. This is the lower-case of a DiagnosticCategory member.

```

Comment:
the generated protocol.d.ts does not have `DiagnosticCategory`.  so I would keep the comment in there. 

One other thing also is we allow extensions  to set it, so it is possible that there are other values here. so a union of string literals would be too restrictive for clients.